pkg/trait/quarkus.go

```diff
@@ -308,7 +308,7 @@ func propagateKitTraits(e *Environment) v1.IntegrationKitTraits {
 		propagate(fmt.Sprintf("integration profile %q", e.IntegrationProfile.Name), e.IntegrationProfile.Status.Traits, &kitTraits, e)
 	}
 
-	propagate(fmt.Sprintf("integration %q", e.Integration.Name), e.Integration.Spec.Traits, &kitTraits, e)
+	propagate(fmt.Sprintf("integration %q", e.Integration.Name), e.Integration.Status.Traits, &kitTraits, e)
```

Comment:
Pardon if this may be a totally wrong question but just looking at the code and from some old memories, since the traits are persisted to the status after they have been applied, what the `quarkus` trait sees in the status are the traits for from the previous reconciliation, isn't it ?

Reply:
Good point. Probably we need to use the "executed trait" that have been used in this reconciliation loop instead of either the `spec` (which represent the user specification) or the `status` (which represent the previous reconciliation loop execution).